src/software/amazon/ion/IonSystem.java

```diff
@@ -204,66 +204,99 @@ public SymbolTable newSharedSymbolTable(String name,
 
 
     /**
+     * <p>
      * Creates an iterator over a stream of Ion text data.
      * Values returned by the iterator have no container.
+     * </p>
+     *
      * <p>
      * The iterator will automatically consume Ion system IDs and local symbol
      * tables; they will not be returned by the iterator.
+     * </p>
+     *
      * <p>
      * If the input source throws an {@link IOException} during iteration, it
      * will be wrapped in an {@link IonException}. See documentation there for
      * tips on how to recover the cause.
+     * </p>
+     *
      * <p>
      * This method is suitable for use over unbounded streams with a reasonable
      * schema.
+     * </p>
+     *
      * <p>
      * Applications should generally use {@link #iterate(InputStream)}
      * whenever possible, since this library has much faster UTF-8 decoding
      * than the Java IO framework.
+     * </p>
+     *
      * <p>
      * Because this library performs its own buffering, it's recommended that
      * you avoid adding additional buffering to the given stream.
+     * </p>
      *
-     * @param ionText a stream of Ion text data.  The caller is responsible for
+     * @param ionText a stream of Ion text data. The caller is responsible for
      * closing the Reader after iteration is complete.
      *
-     * @return a new iterator instance.
+     * @return a new iterator instance. Closing this iterator
+     * will close ionText Reader. The iterator automatically closes when
+     * there are no more values to return.
      *
      * @throws NullPointerException if <code>ionText</code> is null.
      * @throws IonException if the source throws {@link IOException}.
      */
-    public Iterator<IonValue> iterate(Reader ionText);
-
+    public CloseableIterator<IonValue> iterate(Reader ionText);
```

Comment:
This change breaks binary compatibility, and should not be released without a major-version bump.  Java developers expect Jar updates to be drop-in compatible, and this change will break existing code.